crates/oxc_language_server/src/main.rs

```diff
@@ -31,43 +33,26 @@ struct Backend {
     client: Client,
     root_uri: OnceCell<Option<Url>>,
     server_linter: RwLock<ServerLinter>,
+    document_content_cache: DashMap<String, String>,
```

Comment:
I do know now if this could lead to performance problems. We need to keep all content of any open file in the memory.